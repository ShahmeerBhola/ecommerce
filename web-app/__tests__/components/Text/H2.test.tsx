import React from 'react';
import renderer from 'react-test-renderer';
import H2 from '../../../components/Text/H2';

describe('H2 component', () => {
  test('it renders properly with default color', () => {
    const tree = renderer.create(<H2>Header</H2>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders properly with non-default color', () => {
    const tree = renderer.create(<H2 color="red-200">Header</H2>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders properly with className prop', () => {
    const tree = renderer.create(<H2 className="my-custom-class">Header</H2>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
