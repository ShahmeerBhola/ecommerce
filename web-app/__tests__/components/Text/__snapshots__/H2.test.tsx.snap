// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H2 component it renders properly with className prop 1`] = `
<h2
  className="font-primary font-bold text-xl uppercase text-gray-400 my-custom-class"
>
  Header
</h2>
`;

exports[`H2 component it renders properly with default color 1`] = `
<h2
  className="font-primary font-bold text-xl uppercase text-gray-400"
>
  Header
</h2>
`;

exports[`H2 component it renders properly with non-default color 1`] = `
<h2
  className="font-primary font-bold text-xl uppercase text-red-200"
>
  Header
</h2>
`;
